artifact-manifest.md

```diff
@@ -0,0 +1,551 @@
+- **Manifests** are dependent references to other artifacts that enhance the content, such as a Notary v2 signature or an SBoM. These dependencies are *unknown* by the original artifact as they are added at a later time. A registry would need to index these references as registry apis would request all content related to the source artifact.
+
+### Blobs Collection
+
+All `blobs` are considered to be hard dependencies that must be resolvable within a registry. An artifact is considered invalid if the manifest blobs are not resolvable. Registries MAY implement de-duping, using ref-counting to assure at least one copy of the blob is resolvable for any given `oci.artifact.manifest`. OCI Artifact blobs are generalizations of the OCI Image Spec layers definition.
+
+### Manifests Collection
+
+Alternate names:
+
+- `dependent-upon`
+- `parents`
+- `enhances`
+
+The `manifests` collection is an optional collection of OCI Content Descriptors that reference other artifacts. The artifact is said to enhance the dependent artifacts by adding additional content. The content may be added after the initial content was created or pushed to a registry. By supporting additional content, the referenced artifact can be enhanced without having to change the referenced artifacts manifest, digest or tag.
+
+Extension artifacts MAY be stored with or without tags. All extension artifacts MUST be stored in the same repository as the artifact they are extending (linking).
+
+Examples include:
+
+- Notary v2 signatures
+- SBoM documents
+- Artifact Meta-data
+
+## Linked Artifacts API
+
+Returns all artifacts that are linked a given manifest digest. Linked artifact requests are scoped to a repository, ensuring access tokens for the repository can be used as authorization for the linked artifacts.
+
+### Request All Linked Artifacts
+
+**OPTION A**
+
+Part of the v2 distribution API set:
+
+```rest
+GET /v2/{repository}/manifests/{digest}/links?n=10
+```
+
+**OPTION B**
+
+Defined as an extension `/oci/artifacts/v1/`
+
+```rest
+GET /oci/artifacts/v1/{repository}/manifests/{digest}/links?n=10
+```
+
+### Request Artifacts of a specific media type
+
+```rest
+GET /v2/{repository}/manifests/{digest}/links?artifactType={artifactType}&n=10
+
+GET /v2/{repository}/manifests/{digest}/links?artifactType=application/vnd.oci.notary.v2.config+json&n=10
+
+GET /oci/artifacts/links/v2/{repository}/manifests/{digest}/links?artifactType=application/vnd.oci.notary.v2.config+json&n=10
+
+# or:
+
+GET /v2/{repository}/manifests/{digest}/manifests?artifactType={artifactType}&n=10
+
+GET /v2/{repository}/manifests/{digest}/manifests?artifactType=application/vnd.oci.notary.v2.config+json&n=10
+
+GET /oci/artifacts/links/v2/{repository}/manifests/{digest}/manifests?artifactType=application/vnd.oci.notary.v2.config+json&n=10
+```
+
+### Linked Artifact List API results
+
+A summary paged result of manifests, including the descriptor and annotations.
```

Comment:
The major downside of both of these implementation proposals is the need to return the entire contents of all the manifests being requested; perhaps as an alternative the API should return just the digest(s) of the artifact manifest(s) being returned, and then those manifest(s) could be requested via the standard manifest API? It would require a few more round trips, but has the benefit of making use of the existing manifest endpoint (for security, logging, etc), and not requiring loading of all manifest contents when paginating (which could be expensive if manifest contents are in a blob store).

Reply:
I debated adding the linked/listing topic here. I added it for completion of the scenario explanation, but yes. The details for the API/response is a good conversation I was hoping to capture here: https://github.com/opencontainers/distribution-spec/pull/232

Reply:
> perhaps as an alternative the API should return just the digest(s)

For a variety of reasons, we should probably return an entire [descriptor](https://github.com/opencontainers/image-spec/blob/master/descriptor.md) instead of just the digest, so clients know how to handle the digest.

Reply:
>jonjohnsonjr 4 hours ago 
> perhaps as an alternative the API should return just the digest(s)
>For a variety of reasons, we should probably return an entire descriptor instead of just the digest, so clients know how to handle the digest.

For the listing API discussions, I'd suggest we centralize these under the [Listing API Requirements #229](https://github.com/opencontainers/distribution-spec/pull/229). I responded [here](https://github.com/opencontainers/distribution-spec/pull/229#issuecomment-781750955)

Reply:
The `manifest-references-api.md` in this PR has the details. It does return a paged manifest list as different artifact types may use different elements in the manifest. 
Some may be able to identify which artifact to pull based on the `manifest.artifactType`, while a collection of signatures may use an annotation specific to the specific signature format. 
Using just the descriptor doesn't provide enough details to minimize the round trips of pulling each manifest, which means there are that many more round trips.

We hope the optional filtering by `artifactType` will help narrow the scope.